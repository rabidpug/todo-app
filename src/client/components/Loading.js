// @flow

import { Icon, Spin, } from 'antd';

import React from 'react';
import styles from './Loading.scss';

const antIcon = ( <Icon
  spin
  style={ { fontSize: 24, } }
  type='loading' /> );
const Loading: Function = () => (
  <Spin
    className={ styles.spinner }
    delay={ 500 }
    indicator={ antIcon }
    size='large'
    tip='Loading...'
    wrapperClassName={ styles.loadingContent }>
    <div />
  </Spin>
);

export default Loading;
