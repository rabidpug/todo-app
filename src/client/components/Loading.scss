.loadingContent {
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
}

.spinner {
  max-height: inherit !important;
  min-height: 250px;
}
