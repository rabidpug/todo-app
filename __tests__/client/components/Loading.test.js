import Loading from 'Components/Loading';
import React from 'react';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'Loading', () => {
    it(
      'Should render the Loading', () => {
        const wrapper = shallow( <Loading /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
