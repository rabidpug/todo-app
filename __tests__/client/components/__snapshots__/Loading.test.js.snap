// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Should render the Loading 1`] = `
<Spin
  className="spinner"
  delay={500}
  indicator={
    <Icon
      spin={true}
      style={
        Object {
          "fontSize": 24,
        }
      }
      type="loading"
    />
  }
  prefixCls="ant-spin"
  size="large"
  spinning={true}
  tip="Loading..."
  wrapperClassName="loadingContent"
>
  <div />
</Spin>
`;
